libsolidity/ast/Types.cpp

```diff
@@ -345,7 +345,7 @@ TypePointer IntegerType::binaryOperatorResult(Token::Value _operator, TypePointe
 		return TypePointer();
 
 	// All integer types can be compared
-	if (Token::isCompareOp(_operator))
+	if (Token::isCompareOp(_operator) || Token::isShiftOp(_operator))
```

Comment:
Do we want to support shifting for addresses? I would say it is only valid for integer and bytes types.